MdeModulePkg/Include/Protocol/PciDoe.h

```diff
@@ -0,0 +1,73 @@
+/** @file
+  EFI PCI I/O Protocol provides the basic Memory, I/O, PCI configuration,
+  and DMA interfaces that a driver uses to access its PCI controller.
+
+  Copyright (c) 2006 - 2018, Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#ifndef PCI_PROTOCOL_DOE_H_
+#define PCI_PROTOCOL_DOE_H_
+
+typedef struct _EFI_PCI_IO_DOE EFI_PCI_IO_DOE;
+
+/**
+  Check if a specific Vendor ID and Data Object Type is supported
+
+  @param  This                  A pointer to the EFI_PCI_IO_PROTOCOL instance.
+  @param  VendorId              The Vendor ID to check against
+  @param  Type                  The Data Object Type to check against
+
+  @retval EFI_SUCCESS           The operation on the PCI controller's attributes was completed.
+  @retval EFI_INVALID_PARAMETER One or more parameters are invalid.
+  @retval EFI_UNSUPPORTED       DOE is not supported or the VendorId and Type are not supported
+
+**/
+typedef
+EFI_STATUS
+(EFIAPI *EFI_PCI_IO_DOE_TYPE_CHECK)(
+  IN EFI_PCI_IO_PROTOCOL              *This,
+  IN     UINTN                        VendorId,
+  IN     UINTN                        Type
+  );
+
+/**
+  Performs an send or receive operation on a DOE Mailbox
+
+  @param  This                  A pointer to the EFI_PCI_IO_PROTOCOL instance.
+  @param  Count                 The number of bytes to use in the write operation or the maximum number
+                                of bytes to read for a read operation.
+                                For read operations it is set to the number of bytes read.
+  @param  Buffer                For read operations, the destination buffer to store the results. For write
+                                operations, the source buffer to write data from.
+
+  @retval EFI_SUCCESS           The operation on the PCI controller's attributes was completed.
+  @retval EFI_INVALID_PARAMETER One or more parameters are invalid.
+  @retval EFI_UNSUPPORTED       DOE is not supported
+
+**/
+typedef
+EFI_STATUS
+(EFIAPI *EFI_PCI_IO_DOE_MAILBOX)(

```

Comment:
Usually, we define SEND and RECEIVE with different type.

For send, `IN UINTN Count, IN VOID *Buffer`
For receive, `IN OUT UINTN *Count, IN OUT VOID *Buffer`